Exiled.CustomModules/API/Features/CustomItems/CustomItem.cs

```diff
@@ -0,0 +1,651 @@
+// -----------------------------------------------------------------------
+// <copyright file="CustomItem.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.CustomModules.API.Features.CustomItems
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Reflection;
+
+    using Exiled.API.Features;
+    using Exiled.API.Features.Core;
+    using Exiled.API.Features.Core.Interfaces;
+    using Exiled.API.Features.Items;
+    using Exiled.API.Features.Pickups;
+
+    using UnityEngine;
+
+    /// <summary>
+    /// A class to easily manage item behavior.
+    /// </summary>
+    public abstract class CustomItem : TypeCastObject<CustomItem>, IAdditiveBehaviour
+    {
+        /// <inheritdoc cref="ItemManager"/>
+        internal static readonly Dictionary<Item, CustomItem> ItemsValue = new();
+
+        /// <inheritdoc cref="PickupManager"/>
+        internal static readonly Dictionary<Pickup, CustomItem> PickupValue = new();
+
+        private static readonly List<CustomItem> Registered = new();
+
+        /// <summary>
+        /// Gets a <see cref="List{T}"/> which contains all registered <see cref="CustomItem"/>'s.
+        /// </summary>
+        public static IEnumerable<CustomItem> List => Registered;
+
+        /// <summary>
+        /// Gets all Items and their respective <see cref="CustomItem"/>.
+        /// </summary>
+        public static IReadOnlyDictionary<Item, CustomItem> ItemManager => ItemsValue;
+
+        /// <summary>
+        /// Gets all Pickups and their respective <see cref="CustomItem"/>.
+        /// </summary>
+        public static IReadOnlyDictionary<Pickup, CustomItem> PickupManager => PickupValue;
+
+        /// <summary>
+        /// Gets all pickups belonging to a <see cref="CustomItem"/>.
+        /// </summary>
+        public static HashSet<Pickup> CustomPickups => PickupManager.Keys.ToHashSet();
+
+        /// <summary>
+        /// Gets all items belonging to a <see cref="CustomItem"/>.
+        /// </summary>
+        public static HashSet<Item> CustomItems => ItemManager.Keys.ToHashSet();
```

Comment:
We may have to find another name for that.

Reply:
Check the new name